### PR TITLE
Add trigger providers extension point

### LINKED ISSUE #4: Add support for "triggers"
In my server I have a need to perform automatic model updates in response to changes that are made by client applications, to maintain the integrity of certain relationships between objects in my models.
This is akin to the server-side "triggers" commonly implemented in relational database systems.

So, alongside the extension points for custom command providers and validation providers, it would be nice to have an extension point for trigger providers.
These should be given the JSON Patches resulting from `PATCH` requests on the `/api/v2/models` endpoint and provide additional changes to be performed, ideally as further JSON Patches or transactions (as in custom command providers). These new changes then should be applied to the model and incorporated in the same undoable operation as the original changes that triggered them.

As a first iteration of the feature, when there are trigger providers available, an edit should be performed in a transaction (regardless of any custom command providers) so that the transaction context is available to capture trigger changes. Also, because triggers update the model, the changes that they effect should be run through the trigger mechanism recursively. This process iterates until no trigger providers have any further changes to make.

### PR DESCRIPTION
**Pull request by cdamus**
- add extension point for trigger providers
- add example of a trigger for the `Task::duration` attribute
- integrate the example trigger plug-in in the example server
- minor refactoring of command provider for transaction API used by both extension points

Fixes #4.

Contributed on behalf of STMicroelectronics.

### CODE REVIEW

**Review by eneufeld — COMMENTED** [review]
This looks fine, can you rebase this on #5 so that the module loading works the same.

**Comment by cdamus** [comment]
Thanks, @eneufeld . Commit a69ffa733a688a0e0228672749ad4ef1f7a43847 is rebased on #5.

File: packages/modelserver-node/src/client/model-server-client.ts [hunk 1]
```diff
@@ -327,6 +337,10 @@ export class InternalModelServerClient implements InternalModelServerClientApi {
     }
 }
 
+interface NestedEditContext {
```
Comment (eneufeld) [inline packages/modelserver-node/src/client/model-server-client.ts:340]: Should we add a short line describing what this is and why this is needed?

File: packages/modelserver-node/src/client/model-server-client.ts [hunk 2]
```diff
@@ -383,6 +398,14 @@ class DefaultTransactionContext implements TransactionContext {
     }
```
Comment (eneufeld) [inline packages/modelserver-node/src/client/model-server-client.ts:399]: I like the doc on performTriggers, can we add a short line to all other public methods? (I know it is a bit out of context on this pr)

**Review by eneufeld — COMMENTED** [review]
I know we don't have tests yet, would it still make sense maybe adding an initial test for the triggers? if we don't start we won't get any ;-)

**Comment by cdamus** [comment]
Hi @eneufeld . Commit 5ba2bb644f9c6d449f1d2bd1bd654e6d8d363dcd should address your comments, including adding a very few initial tests for the `TriggerProviderRegistry`.

**Review by eneufeld — APPROVED** [review]
Thank you!

**Comment by cdamus** [comment]
Commit e3a9534e5aea717b1a9fc072d4fe4f5d0ec546dc rebases onto the latest `main` branch to resolve merge conflicts.

**Pull request merged**